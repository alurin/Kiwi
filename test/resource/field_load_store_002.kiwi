#!/usr/bin/env kiwi
/* Test script for object fields */

#!/usr/bin/env kiwi

class A {
    int @a;

    void set(int $a) {
        @a = $a;
    }

    void write() {
        print "Value: ";
        print @a;
        print "\n";
    }
}

class B {
    int @c;
    int @b;

    inherit A;

    void setB(int $b, int $c) {
        @b = $b;
        @c = $c;
    }

    void writeB() {
        print "Value: ";
        print @b;
        print "\n";

        print "Value: ";
        print @c;
        print "\n";
    }
}

/// Main function
void main() {
    auto $a = new B();
    $a.set(10);
    $a.setB(11, 12);
    $a.write();
    $a.writeB();
    $a.setB(13, 14);
    $a.set(15);
    $a.writeB();
    $a.write();
}
