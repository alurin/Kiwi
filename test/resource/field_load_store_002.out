Value: 10
Value: 11
Value: 12
Value: 13
Value: 14
Value: 15
